express.js

```diff
@@ -1,63 +1,108 @@
 const express = require('express');
 const app = express();
+require('dotenv').config();
+const { v4: uuidv4 } = require('uuid');
 const { readUsers, writeUsers } = require('./helpers.js');
-
 app.use(express.json());
-const PORT = 3054;
+app.set('PORT', process.env.PORT || 3080);
+
+   // TODO add id as first level obecht key in json
 
-app.get('/users', (req, res) => {
-    let info = readUsers();
-    res.json(info.users);
+app.get('/users/:id', (req, res) => {
+    let users = readUsers();
+    let param = req.params.id;
+
+    let user = users[param];
+    if (!user) {
+        return res.status(404).json("User not found");
+    }
+
+    res.json(user);
 });
 
+
+// TODO user can have email which is unique
+// TODO userId must be UUID
+// user add operation needs to recieve metadata and then spread
+
+// get operation with id and email needs to be o(1)
+
 app.post('/users', (req, res) => {
-    let info = readUsers();
-    console.log(info);
-    const { name, surname } = req.body;
+    let users = readUsers();
+    const { name, surname, email, meta } = req.body;
+    
+    if (Object.values(users).some(u => u.email === email)) {
```

Comment:
Your get by email function is not `o(1)` is was a requirement 
`// get operation with id and email needs to be o(1)`